### PR TITLE
Celery

### PR DESCRIPTION
**Pull request by bbelderbos**

### CODE REVIEW

File: project/app/mongo_db.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+load_dotenv()
+
+client = MongoClient(os.getenv("MONGO_URI"),
+                     connectTimeoutMS=200,
+                     retryWrites=True)
+db = client.music_library
+songs = db.songs
+
+Song = dict[str, Union[float, str, int]]
```
[the remark this answers is not in the document]
Comment (bbelderbos) [reply project/app/mongo_db.py:15]: Btw I should use an IDE, so much easier to determine the typing of nested objects!
https://pybit.es/articles/code-better-with-type-hints-part-2/

File: project/app/api/spot.py [hunk 2]
```diff
@@ -62,22 +58,49 @@ def get_track_features(artist: str, name: str):
...
+
+    # TODO: how to prevent we kick off two download tasks?
+    # so far we just check the DB for songs from artist
+    # as a criteria of "job done"
+    songs = retrieve_song_features(artist)
+
+    if songs.count() > 0:
+        return [_serialize(song) for song in songs]
+
+    job = retrieve_and_cache_song_features.delay(artist)
```
[the remark this answers is not in the document]
Comment (jengroff) [reply project/app/api/spot.py:91]: whoa. just reviewing this all now.

File: project/app/mongo_db.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import os
+from typing import Union
+
+from pymongo import MongoClient
+from dotenv import load_dotenv
+
+load_dotenv()
+
+client = MongoClient(os.getenv("MONGO_URI"),
```
[the remark this answers is not in the document]
Comment (jengroff) [reply project/app/mongo_db.py:9]: No we don't really need MongoDB; I was playing around a bit and also like the bulk insert method, which is so easy with mongo.

File: project/app/mongo_db.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jengroff) [reply project/app/mongo_db.py:15]: ha! I can't see you giving up vim.

File: project/app/mongo_db.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+Song = dict[str, Union[float, str, int]]
+ArtistSongs = list[Song]
+
+
+def bulk_insert_songs(song_features: ArtistSongs) -> int:
+    songs.insert_many(song_features)
+    return songs.count_documents({})
+
+
+def retrieve_song_features(artist: str) -> list[Song]:
```
[the remark this answers is not in the document]
Comment (jengroff) [reply project/app/mongo_db.py:24]: Yes, 3.9. Did I not have` from typing import List` ?

**Pull request merged**